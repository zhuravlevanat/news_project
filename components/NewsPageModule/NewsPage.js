import React from "react";
import Layout from "../LayoutModule/Layout";
import NewsPost from "../PostModule/NewsPost";
import List from "@material-ui/core/List";
import Spinner from "../helpers/Spinner";
import Error from "../helpers/Error";

const NewsPage = ({ news, loading, error }) => {
    console.log('news', news);
    return (
        <Layout >
            {loading
                ? <Spinner/>
                : <List>{news.map((item, index) => (<NewsPost key={index} item={item}/>))}</List>}
            {error && <Error />}
        </Layout>
    )
};

export default NewsPage;
