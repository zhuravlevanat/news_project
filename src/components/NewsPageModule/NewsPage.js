import React from "react";
import Layout from "../LayoutModule/Layout";
import NewsList from "./NewsList";
import Error from "../helpers/Error";

const NewsPage = ({ news, loading, error }) => (
    <Layout >
        {error ? <Error /> : <NewsList news={news} loading={loading}/>}
    </Layout>
);

export default NewsPage;
