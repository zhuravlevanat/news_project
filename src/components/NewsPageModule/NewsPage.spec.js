import React from "react";
import { shallow } from 'enzyme';
import NewsPage from "./NewsPage";

describe('NewsPage', () => {
    const props = {
        news: [],
        error: false,
    };
    describe('News page initial', () => {
        const newsPage = shallow(<NewsPage {...props} />);
        it('renders properly', () => {
            expect(newsPage).toMatchSnapshot();
        });

        it('render initial', () => {
            expect(newsPage.find('Error')).toHaveLength(0);
            expect(newsPage.find(`NewsList`)).toHaveLength(1);
        });
    });
    describe('News page with error', () => {
        const nextProps = {
            ...props,
            error: true,
        };
        const newsPage = shallow(<NewsPage {...nextProps} />);

        it('renders properly', () => {
            expect(newsPage).toMatchSnapshot();
        });

        it('render error', () => {
            expect(newsPage.find('Error')).toHaveLength(1);
            expect(newsPage.find(`NewsList`)).toHaveLength(0);
        });
    });
});
