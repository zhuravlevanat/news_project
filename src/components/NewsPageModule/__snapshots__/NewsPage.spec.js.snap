// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsPage News page initial renders properly 1`] = `ShallowWrapper {}`;

exports[`NewsPage News page with error renders properly 1`] = `ShallowWrapper {}`;
